### PR TITLE
SLM-110 long names

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: integration_tests/integration/barcode/createNewContact.spec.ts [hunk 1]
```diff
@@ -19,6 +19,14 @@ context('Create New Contact Page', () => {
...
   })
 
+  it.only('should find a prison when searching with HMP prefix', () => {
+    const createNewContactPage = FindRecipientByPrisonNumberPage.goToPage().submitWithValidPrisonNumber()
+
+    createNewContactPage.enterAValidPrisonerName().typeAheadHMPValidPrison().submitForm(ReviewRecipientsPage)
+
+    Page.verifyOnPage(ReviewRecipientsPage)
+  })
+
```
Comment (mikehalmamoj) [inline integration_tests/integration/barcode/createNewContact.spec.ts:29]: Missing test from previous PR

File: server/routes/barcode/newContactValidator.ts [hunk 2]
```diff
@@ -5,17 +5,19 @@ const PRISONER_NAME_PATTERN = /^[a-zA-Z '`-]+$/
...
+    errors.push({ href: '#prisonerName', text: 'Name can have a maximum length of 60 characters.' })
   } else {
     req.body.prisonerName = req.body.prisonerName.trim()
     if (!PRISONER_NAME_PATTERN.test(req.body.prisonerName)) {
       errors.push({ href: '#prisonerName', text: 'Enter names that only use letters, not numbers or symbols.' })
     }
   }
+  if (!req.body.prisonId) {
+    errors.push({ href: '#prisonId', text: 'Select a prison name' })
+  }
```
Comment (mikehalmamoj) [inline server/routes/barcode/newContactValidator.ts:20]: Separate fix for SLM-92 - validate in order fields appear on screen so the error messageg are in the correct order.

**Review by joelstobart-moj — DISMISSED** [review]
LGTM

File: integration_tests/integration/barcode/createNewContact.spec.ts [hunk 3]
```diff
@@ -19,6 +19,14 @@ context('Create New Contact Page', () => {
     Page.verifyOnPage(ReviewRecipientsPage)
   })
 
+  it.only('should find a prison when searching with HMP prefix', () => {
```
Comment (nathanrussell-moj-digital) [inline integration_tests/integration/barcode/createNewContact.spec.ts:22]: What does the `it.only` do?

Comment (mikehalmamoj) [reply integration_tests/integration/barcode/createNewContact.spec.ts:22]: Pants. It only runs that test locally. I'll remove, good spot!

File: server/routes/barcode/newContactValidator.test.ts [hunk 4]
```diff
@@ -83,4 +83,15 @@ describe('newContactValidator', () => {
       ])
     })
   })
+  it(`should not allow names longer than 60 characters`, () => {
```
Comment (nathanrussell-moj-digital) [inline server/routes/barcode/newContactValidator.test.ts:86]: I'm OK with this, as I'm sure any sensible pragmatic person would be, but I did notice KH squirming in the standup when you showed it. You just know he's going to say "you can't exclude people with really long names" !

Comment (mikehalmamoj) [reply server/routes/barcode/newContactValidator.test.ts:86]: When he finds a name longer than 60 chars I'll listen, but pragmatism must prevail if we ever want to put this live. (Sorry I know I'm preaching to the choir!)

File: server/services/barcode/CreateBarcodeService.ts [hunk 5]
```diff
@@ -111,21 +112,43 @@ export default class CreateBarcodeService {
...
+    // Calculate how to split the name into 2 lines
+    let name1length = Math.min(30, name.length - 4)
+    let lineBreakChar = '-'
+    const naturalLineBreak = name.substring(26, 34).indexOf(' ')
+    if (naturalLineBreak !== -1) {
+      lineBreakChar = ''
+      name1length = naturalLineBreak + 26
+    }
+    // Return address including 2 lines of names but without street
+    const name1 = `${name.substring(0, name1length)}${lineBreakChar}`.trim()
```
Comment (nathanrussell-moj-digital) [inline server/services/barcode/CreateBarcodeService.ts:150]: KH is going to hate the cases where we couldnt find a suitable natural break and ended up putting a hyphen in the middle of someones name! 🤣

**Review by nathanrussell-moj-digital — APPROVED** [review]

**Pull request merged**